Source/com/drew/imaging/png/PngMetadataReader.java

```diff
@@ -323,6 +327,21 @@ private static void processChunk(@NotNull Metadata metadata, @NotNull PngChunk c
             PngDirectory directory = new PngDirectory(PngChunkType.sBIT);
             directory.setByteArray(PngDirectory.TAG_SIGNIFICANT_BITS, bytes);
             metadata.addDirectory(directory);
+        } else if (chunkType.equals(PngChunkType.eXIF)) {
+            try {
+                ExifTiffHandler handler = new ExifTiffHandler(metadata, null);
+                RandomAccessStreamReader reader = new RandomAccessStreamReader(new ByteArrayInputStream(bytes));
+                int tiffHeaderOffset = 0;
+                new TiffReader().processTiff(reader, handler, tiffHeaderOffset);
+            } catch (TiffProcessingException e) {
+                PngDirectory directory = new PngDirectory(PngChunkType.eXIF);
+                directory.addError("Exception processing the PNG image exif information.");
+                metadata.addDirectory(directory);
+            } catch (IOException e) {
+                PngDirectory directory = new PngDirectory(PngChunkType.eXIF);
+                directory.addError("Exception reading the PNG image exif information.");
+                metadata.addDirectory(directory);
+            }
```

Comment:
I think it's better to just propagate the original exception message, as it will contain the most useful information for diagnosis. It'll be in the Exif PNG chunk so the origin should be clear.

```suggestion
            } catch (TiffProcessingException | IOException e) {
                PngDirectory directory = new PngDirectory(PngChunkType.eXIF);
                directory.addError(ex.getMessage());
                metadata.addDirectory(directory);
            }
```